src/saml.py

```diff
@@ -45,26 +30,47 @@ def __init__(self, charm_state: CharmState):
 
         Args:
             charm_state: The state of the charm that the Saml instance belongs to.
+
+        Raises:
+            CharmConfigInvalidError: if the certificate validation fails.
         """
         self._charm_state = charm_state
+        try:
+            if not self._is_certificate_valid:
+                raise CharmConfigInvalidError(
+                    (
+                        f"The certificate from {self._charm_state.metadata_url} "
+                        "doesn't match the provided fingerprint"
+                    )
+                )
+        except TimeoutError as ex:
+            raise CharmConfigInvalidError(
+                f"Error while validating certificate from {self._charm_state.metadata_url}"
+            ) from ex
 
-    @property
-    def entity_id(self) -> str:
-        """Return entity_id config.
+    @cached_property
```

Comment:
Out of curiosity: if the certificate is updated, how this would be re-calculated?

Reply:
I'm afraid that's yet to be determined

Reply:
I see, this could be addressed in a further story, perhaps?

Reply:
Yees, we'll certainly need a way to reload this